internal/config/parser/viper.go

```diff
@@ -0,0 +1,18 @@
+// This package provides viper parser.
+//
+// Due to bug in v1 API, it is not possible to set custom parser in the global viper
+// package. For this reason, all parsing must be done in a separate package where
+// viper is initialized with custom StringReplacer.
+//
+// https://github.com/spf13/viper/pull/870
+//
+package parser
+
+import "github.com/spf13/viper"
+
+var Viper *viper.Viper
+
+func init() {
+	Viper = viper.NewWithOptions(viper.EnvKeyReplacer(camelCaseReplacer{}))
+	Viper.AutomaticEnv()
+}
```

Comment:
This is the reason why viper must be extracted into package (because it is used from other `init` function - ordering issue) and because `SetEnvKeyReplacer` dos not accept `StringReplacer` interface so it's literally unusable.

Reply:
I stand convinced viper is not a great tool xD